packages/vm/src/compatability.rs

```diff
@@ -40,11 +42,46 @@ pub fn check_wasm(wasm_code: &[u8]) -> Result<()> {
             ));
         }
     };
+    check_wasm_memories(&module)?;
     check_wasm_exports(&module)?;
     check_wasm_imports(&module)?;
     Ok(())
 }
 
+fn check_wasm_memories(module: &Module) -> Result<()> {
+    let section = match module.memory_section() {
+        Some(section) => section,
+        None => {
+            return make_validation_err("Wasm contract doesn't have a memory section".to_string());
+        }
+    };
+
+    let memories = section.entries();
+    if memories.len() != 1 {
+        return make_validation_err("Wasm contract must contain exactly one memory".to_string());
+    }
+
+    for memory in memories {
+        // println!("Memory: {:?}", memory);
+        let limits = memory.limits();
+
+        if limits.initial() > MEMORY_LIMIT {
+            return make_validation_err(format!(
+                "Wasm contract memory's minimum must not exceed {} pages.",
+                MEMORY_LIMIT
+            ));
+        }
+
+        if limits.maximum() != None {
+            return make_validation_err(
+                "Wasm contract memory's maximum must be unset. The host will set it for you."
```

Comment:
I thought we allowed fixed maximums, as long as they were less than MEMORY_LIMIT

Reply:
This question is a bit open to me. First I thought we can let them pass, now I'm not sure if we should ever start supporting them.

If we keep it that way, we're more flexible on the VM side to choose an approriate limit. If contracts start setting somthing as maximum, we need more coordination which value is valid.

Reply:
Okay, fair to leave it like this with a comment on behavior and revisit it if anyone starts uploading valid contracts with fixed maximums.